[10_6_X] Backport concurrent GEN utilities

colizz opened the pull request:
#### PR description:

Backports of all recently developed/modified concurrent GEN utilities to 10_6_X, to benefit UL event processing.

The PR includes the following changes

1. #33985 
2. #34146
3. #34173 
4. #34437 
5. #34525
6. #34436 (this is the change to the analysis code)
7. Backport the changes on cfi/cff files in `Configuration/Generator/python` to enable concurrent GEN. This includes several items (see details in https://github.com/cms-sw/cmssw/pull/34527#issuecomment-893693894):
    - Backport MG and Pythia config #29554
    - Backport Herwig config changes #32337, #33599
    - Big backport from #34710 following the same four-step rule. It also includes the backport of #29305, #33516, #34463 with the same goal (these are never backported to 10_6_X)
8. #34804
9. #35072

Small modifications are made, if necessary, to adapt the 10_6_X code. Please see details in each commit message.


#### PR validation:

Validations are made on specific UL processes to see the GEN utility works (some recipes below). 
We also validate the physics result on the vast UL samples under 10_6_X, which are reported to GEN group. Please see [slides here](https://indico.cern.ch/event/1058787/contributions/4449901/attachments/2283917/3881832/21.07.19_GEN_Update%20on%20multi-threading%20validation.pdf).

1.
  
validated on [`JME-RunIISummer20UL17wmLHEGEN-00001`](https://cms-pdmv.cern.ch/mcm/requests?prepid=JME-RunIISummer20UL17wmLHEGEN-00001) (MG5+Herwig7)
```shell
cd $CMSSW_BASE/src
curl -s -k https://cms-pdmv.cern.ch/mcm/public/restapi/requests/get_fragment/JME-RunIISummer20UL17wmLHEGEN-00001 --retry 3 --create-dirs -o Configuration/GenProduction/python/JME-RunIISummer20UL17wmLHEGEN-00001-fragment.py
scram b -j8
cd ../..
cmsDriver.py Configuration/GenProduction/python/JME-RunIISummer20UL17wmLHEGEN-00001-fragment.py --python_filename JME-RunIISummer20UL17wmLHEGEN-00001_1_cfg.py --eventcontent RAWSIM,LHE --customise Configuration/DataProcessing/Utils.addMonitoring --datatier GEN,LHE --fileout file:JME-RunIISummer20UL17wmLHEGEN-00001.root --conditions 106X_mc2017_realistic_v6 --beamspot Realistic25ns13TeVEarly2017Collision --step LHE,GEN --geometry DB:Extended --era Run2_2017 --mc -n 20 --nThreads 4 \
 --customise_commands="process.externalLHEProducer.generateConcurrently = cms.untracked.bool(True); process.externalLHEProducer.postGenerationCommand = cms.untracked.vstring('mergeLHE.py', '-i', 'thread*/cmsgrid_final.lhe', '-o', 'cmsgrid_final.lhe')"
```

2&3: 

validated on Pythia8+Tauola workflow
```shell
cmsDriver.py GGToHtautau_13TeV_pythia8_Tauola_taupinu_cff --conditions auto:run2_mc -s GEN --datatier GEN --beamspot Realistic8TeVCollision -n 20 --relval 250000,5000 --eventcontent RAWSIM --fileout file:step1.root --nThreads 4 \
 --customise_commands="from GeneratorInterface.Core.ExternalGeneratorFilter import ExternalGeneratorFilter; process.generator = ExternalGeneratorFilter(process.generator, _external_process_components_=cms.vstring('HepPDTESSource'))"
```

4:

validated on [`B2G-RunIISummer19UL17wmLHEGEN-00001`](https://cms-pdmv.cern.ch/mcm/requests?prepid=B2G-RunIISummer19UL17wmLHEGEN-00001) (MG5+Pythia8)

```shell
cd $CMSSW_BASE/src
curl -s -k https://cms-pdmv.cern.ch/mcm/public/restapi/requests/get_fragment/B2G-RunIISummer19UL17wmLHEGEN-00001 --retry 3 --create-dirs -o Configuration/GenProduction/python/B2G-RunIISummer19UL17wmLHEGEN-00001-fragment.py
scram b -j8
cd ../..
cmsDriver.py Configuration/GenProduction/python/B2G-RunIISummer19UL17wmLHEGEN-00001-fragment.py --python_filename B2G-RunIISummer19UL17wmLHEGEN-00001_1_cfg.py --eventcontent RAWSIM,LHE --customise Configuration/DataProcessing/Utils.addMonitoring --datatier GEN,LHE --fileout file:B2G-RunIISummer19UL17wmLHEGEN-00001.root --conditions 106X_mc2017_realistic_v6 --beamspot Realistic25ns13TeVEarly2017Collision --step LHE,GEN --geometry DB:Extended --era Run2_2017 --mc -n 20 --nThreads 4 \
 --customise_commands="process.externalLHEProducer.scriptName = cms.FileInPath('GeneratorInterface/LHEInterface/data/run_generic_tarball_cvmfs_madgraphLO_multithread.sh')"
```

5. See validation in the original PR.

6.

validated on Pythia8+Tauola and process till the NanoGEN step

```shell
cmsDriver.py GGToHtautau_13TeV_pythia8_Tauola_taupinu_cff --conditions auto:run2_mc -s GEN,NANOGEN --datatier NANOAODSIM --beamspot Realistic8TeVCollision -n 20 --relval 250000,5000 --eventcontent NANOAODSIM --fileout file:nanogen.root \
 --customise_commands="from GeneratorInterface.Core.ExternalGeneratorFilter import ExternalGeneratorFilter; process.generator = ExternalGeneratorFilter(process.generator, _external_process_components_=cms.vstring('HepPDTESSource'))"
echo "Events->Show(10)" | root -l nanogen.root
```
See that the `GenJet_partonFlavour` branch is recovered.

7. Validated on all modified/added RelVal workflows. See details in https://github.com/cms-sw/cmssw/pull/34527#issuecomment-893693894

8/9. See validation in the original PR.

colizz committed:
Add argument postGenerationCommand to ExternalLHEProducer and provide mergeLHE.py script (10_6_X)
Backport #33985 to 10_6_X, the config files in Configuration/Generator/ not changed

colizz committed:
Allow specification of ES modules in ExternalGeneratorFilter (10_6_X)
Backport #34146 to 10_6_X

colizz committed:
Specify separate directories to run external instances in ExternalGeneratorFilter (10_6_X)
Backport #34173 to 10_6_X
Additional change: replace std::filesystem with boost::filesystem

colizz committed:
Fix the MadGraph5 LO multithread utility (10_6_X)
Backport #34437 to 10_6_X

colizz committed:
Extend the external process wait time for ExternalGeneratorFilter (10_6_X)
Backport #34525 to 10_6_X

colizz committed:
Use the original GEN module name for ExternalGeneratorFilter in recognizing generator types (10_6_X)
Backport #34436 to 10_6_X

cmsbuild commented:
A new Pull Request was created by  @colizz (Congqiao Li) for CMSSW_10_6_X.

It involves the following packages:

- GeneratorInterface/Core (**generators**)
- GeneratorInterface/LHEInterface (**generators**)
- PhysicsTools/JetMCAlgos (**analysis**)
- TopQuarkAnalysis/TopEventProducers (**analysis**)


@SiewYan, @mkirsano, @Saptaparna, @cmsbuild, @GurpreetSinghChahal, @agrohsje, @alberto-sanchez, @santocch can you please review it and eventually sign? Thanks.
@alberto-sanchez, @JyothsnaKomaragiri, @emilbols, @andrzejnovak, @mkirsano this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy, @perrotta you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

Saptaparna commented:
@colizz : which workflows would you like to be tested?

colizz commented:
> @colizz : which workflows would you like to be tested?

Hi @Saptaparna , maybe we can follow https://github.com/cms-sw/cmssw/pull/33516. These include LHE(MG5, Powheg) + Hadronzier(P8, H7) routines.

But we do not specify the use of concurrent utilities in any workflow yet. Is it required to modify the configs in a similar way under 10_6_X?

Saptaparna commented:
@colizz Thanks, followed #33516 I see what you mean.

colizz commented:
Thanks, so as we discussed in GEN meeting and offline, I'll first make a change in RelVal workflows in the master branch. Then I'll backport the change by a follow-up commit in this PR.

qliphy commented:
> Thanks, so as we discussed in GEN meeting and offline, I'll first make a change in RelVal workflows in the master branch. Then I'll backport the change by a follow-up commit in this PR.

@colizz  kindly ping.

colizz commented:
(will come back and finalize this thread by Friday - apologies for the delay)

tvami commented:
test parameters:
- workflow 512,547,548,555,513,551,556,514,552,554,515,516,517,518,519,520

colizz committed:
Update & optimize GEN-only Relval workflows (10_6_X)

Backport #29554 to 10_6_X

colizz committed:
adding configuration files for Matchbox and standalone H7 (10_6_X)

Backport #32337 to 10_6_X

colizz committed:
Renaming cfgs and cffs (10_6_X)

Backport #33599 to 10_6_X

colizz committed:
Backport the changes on cfi/cff files in `Configuration/Generator/python` to enable concurrent GEN

 Big backport from #34710 following the same four-step rule. It also includes the backport of #29305, #33516, #34463 with the same goal (these are never backported to 10_6_X)

cmsbuild commented:
Pull request #34527 was updated. @SiewYan, @mkirsano, @jordan-martins, @chayanit, @bbilin, @wajidalikhan, @Saptaparna, @cmsbuild, @adrianodee, @srimanob, @agrohsje, @kpedro88, @kskovpen, @alberto-sanchez, @santocch, @GurpreetSinghChahal can you please check and sign again.

colizz commented:
Backport the changes on cfi/cff files in `Configuration/Generator/python` to enable concurrent GEN, including:
  - Backport MG and Pythia config changes #29554
  - Backport Herwig config changes #32337, #33599
  - Big backport from #34710 following the same four-step rule. It also includes the backport of #29305, #33516, #34463 with the same goal (these are never backported to 10_6_X)

<details>
  <summary>We validated on all modified RelVal workflows for 10_6_X: [click to expend]</summary>
  
  1. workflow 562.0,537.0,512.0,547.0,515.0,518.0,526.0,529.0,522.0,555.0,1361.17,1361.0,1361.18,1361.181,25211.0,25211.18,25211.17,514.0,523.0,551.0,1360.0,25210.0,548.0,556.0,1362.0,1362.17,1362.18,25212.0,25212.17,25212.18,1363.0,1363.17,1363.18,25213.0,25213.17,25213.18,554.0,517.0,520.0,524.0,527.0,530.0,552.0,516.0,519.0,521.0,525.0,528.0,513.0
  2. workflow 1361.17,1361.0,1361.18,1361.181,25211.0,25211.18,25211.17,514.0,552.0,517.0,520.0,512.0,547.0,513.0,515.0,516.0,519.0,518.0,555.0,551.0,548.0,556.0,554.0,562.0,1370.0,1370.17,1370.18,25215.0,25215.17,25215.18,1362.0,1362.17,1362.18,25212.0,25212.17,25212.18,1363.0,1363.17,1363.18,25213.0,25213.17,25213.18
  3. workflow 505.0,546.0,135.7,2017.7,2018.7,25403.0,25403.17,25403.18,250403.0,250403.17,250403.18,5.7,108.0,109.0,103.0,111.0,135.8,1301.0,1301.17,2017.8,2018.8,503.0,1.0,105.0,135.6,2017.6,2018.6,25409.0,25409.17,25409.18,250408.0,250408.17,250408.18,5.6,117.0,112.0,118.0,113.0,1303.0,1303.17,3.0,123.2,504.0,114.0,4007.0,115.0,135.12,149.0,2017.12,2018.12,116.0,123.1,4008.0,107.0,34.0,204.0,133.0,135.11,2017.11,2018.11,25408.0,25408.17,25408.18,250409.0,250409.17,250409.18,507.0,509.0,510.0,508.0,135.1,135.13,1302.0,1302.181,1302.17,1302.18,2017.1,2017.13,2018.1,2018.13,25402.17,25402.0,25402.18,250202.118,250202.183,250202.182,250402.0,250402.17,250402.1,250402.171,250402.18,250402.181,2.0,5.1,121.0,122.0,506.0,561.0,135.4,1304.18,1304.181,1304.182,1356.0,2017.4,2018.4,25400.0,25400.17,25400.18,250200.1,250200.117,250200.118,250200.182,250400.0,250400.17,250400.18,500200.1,5.4,135.9,2017.9,2018.9,25406.0,25406.17,25406.18,250206.118,250206.182,250406.0,250406.17,250406.18,125.0,135.5,2017.5,2018.5,25405.0,25405.18,25405.17,250405.17,250405.0,250405.18,119.0,560.0,106.0
  4. workflow 570,571,572,573,574,575,576,577

</details>

<details>
  <summary>Appendix: [click to expend]</summary>
  
  1. *modified* cfi/cff files for ExternalLHEProducer

| Config filename | Matched RelVal ID(s) |
| :--- | :--- |
| BulkG_M1200_narrow_2L2Q_LHE_13TeV_cff | 562.0 |
| DYToLL012Jets_5FS_TuneCH3_13TeV_amcatnloFxFx_herwig7_cff | 537.0 |
| DYToll01234Jets_5f_LO_MLM_Madgraph_LHE_13TeV_cff | 512.0,547.0,515.0,518.0,526.0,529.0 |
| DYToll012Jets_5f_LO_MLM_Madgraph_LHE_13TeV_cff | 522.0 |
| DYToll012Jets_5f_NLO_FXFX_Madgraph_LHE_13TeV_cff | 555.0 |
| DarkSUSY_mH_125_mN1_60_mGammaD_10_cT_10_Madgraph_LHE_13TeV_cfi |  |
| GGHZZ4L_JHUGen_Pow_NNPDF30_LHE_13TeV_cff |  |
| GGHZZ4L_JHUGen_Pow_NNPDF30_LHE_13TeV_cfi | 1361.17,1361.0,1361.18,1361.181,25211.0,25211.18,25211.17 |
| GGToH_Pow_LHE_13TeV_cff | 514.0 |
| TTbar012Jets_5f_NLO_FXFX_Madgraph_LHE_13TeV_cff | 523.0,551.0 |
| TTbar012Jets_5f_NLO_FXFX_Madgraph_LHE_13TeV_cfi | 1360.0,25210.0 |
| TTbar_Pow_LHE_13TeV_cff | 548.0,556.0 |
| VBFHZZ4Nu_Pow_NNPDF30_LHE_13TeV_cff |  |
| VBFHZZ4Nu_Pow_NNPDF30_LHE_13TeV_cfi | 1362.0,1362.17,1362.18,25212.0,25212.17,25212.18 |
| VBFHbb_Pow_NNPDF30_LHE_13TeV_cff |  |
| VBFHbb_Pow_NNPDF30_LHE_13TeV_cfi | 1363.0,1363.17,1363.18,25213.0,25213.17,25213.18 |
| VBFToH_Pow_JHU4l_LHE_13TeV_cff | 554.0 |
| VHToH_Pow_LHE_13TeV_cff | 517.0,520.0,524.0,527.0,530.0,552.0 |
| WTolNu01234Jets_5f_LO_MLM_Madgraph_LHE_13TeV_cff | 516.0,519.0,521.0,525.0,528.0 |
| WTolNu012Jets_5f_LO_MLM_Madgraph_LHE_13TeV_cff | 513.0 |


  2. *modified* cfi/cff files for PythiaHadronizerFilter

| Config filename | Matched RelVal ID(s) |
| :--- | :--- |
| DarkSUSY_mH_125_mN1_60_mGammaD_10_cT_10_Madgraph_LHE_13TeV_cfi |  |
| GGHZZ4L_JHUGen_Pow_NNPDF30_LHE_13TeV_cfi | 1361.17,1361.0,1361.18,1361.181,25211.0,25211.18,25211.17 |
| Hadronizer_MgmMatchTune4C_13TeV_madgraph_pythia8_cff |  |
| Hadronizer_MgmMatchTuneCUETP8M1_13TeV_madgraph_pythia8_cff |  |
| Hadronizer_MgmMatchTuneCUETP8M1_13TeV_madgraph_pythia8_taupinu_cff |  |
| Hadronizer_MgmMatchTuneCUETP8M1_13TeV_madgraph_pythia8_taurhonu_cff |  |
| Hadronizer_TuneCP5_13TeV_Hgg_powhegEmissionVeto_pythia8_cff | 514.0 |
| Hadronizer_TuneCP5_13TeV_Htt_powhegEmissionVeto_pythia8_cff | 552.0 |
| Hadronizer_TuneCP5_13TeV_Httpinu_powhegEmissionVeto_pythia8_cff | 517.0 |
| Hadronizer_TuneCP5_13TeV_Httrhonu_powhegEmissionVeto_pythia8_cff | 520.0 |
| Hadronizer_TuneCP5_13TeV_MLM_5f_max4j_LHE_pythia8_cff | 512.0,547.0,513.0 |
| Hadronizer_TuneCP5_13TeV_MLM_5f_max4j_LHE_pythia8_taupinu_cff | 515.0,516.0 |
| Hadronizer_TuneCP5_13TeV_MLM_5f_max4j_LHE_pythia8_taurhonu_cff | 519.0,518.0 |
| Hadronizer_TuneCP5_13TeV_aMCatNLO_FXFX_5f_max2j_max0p_LHE_pythia8_cff | 555.0 |
| Hadronizer_TuneCP5_13TeV_aMCatNLO_FXFX_5f_max2j_max1p_LHE_pythia8_cff | 551.0 |
| Hadronizer_TuneCP5_13TeV_powhegEmissionVeto2p_pythia8_cff | 548.0,556.0 |
| Hadronizer_TuneCP5_13TeV_powhegEmissionVeto_pythia8_cff | 554.0 |
| Hadronizer_TuneCUETP8M1_13TeV_Hgg_powhegEmissionVeto_pythia8_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_Htt_powhegEmissionVeto_pythia8_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_Httpinu_powhegEmissionVeto_pythia8_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_Httrhonu_powhegEmissionVeto_pythia8_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_MLM_5f_max4j_LHE_pythia8_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_MLM_5f_max4j_LHE_pythia8_taupinu_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_MLM_5f_max4j_LHE_pythia8_taurhonu_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_aMCatNLO_FXFX_5f_max1j_max1p_LHE_pythia8_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_aMCatNLO_FXFX_5f_max2j_max0p_LHE_pythia8_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_aMCatNLO_FXFX_5f_max2j_max1p_LHE_pythia8_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_generic_LHE_pythia8_cff | 562.0 |
| Hadronizer_TuneCUETP8M1_13TeV_ggHZZ4L_powhegEmissionVeto_pythia8_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_powhegEmissionVeto2p_pythia8_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_powhegEmissionVeto_2p_HToGG_M125_LHE_py8_cff | 1370.0,1370.17,1370.18,25215.0,25215.17,25215.18 |
| Hadronizer_TuneCUETP8M1_13TeV_powhegEmissionVeto_3p_HToBB_M-125_LHE_pythia8_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_powhegEmissionVeto_3p_HToZZ4nu_M-125_LHE_pythia8_cff |  |
| Hadronizer_TuneCUETP8M1_13TeV_powhegEmissionVeto_pythia8_cff |  |
| VBFHZZ4Nu_Pow_NNPDF30_LHE_13TeV_cfi | 1362.0,1362.17,1362.18,25212.0,25212.17,25212.18 |
| VBFHbb_Pow_NNPDF30_LHE_13TeV_cfi | 1363.0,1363.17,1363.18,25213.0,25213.17,25213.18 |

  3. *modified* cfi/cff files for PythiaGeneratorFilter

| Config filename | Matched RelVal ID(s) |
| :--- | :--- |
| ADDMonoJet_13TeV_d3MD3_TuneCUETP8M1_cfi |  |
| ADDMonoJet_13TeV_d3MD3_cfi |  |
| ADDMonoJet_8TeV_d3MD3_TuneCUETP8M1_cfi |  |
| DYToLL_M-50_13TeV_pythia8_cff | 505.0,546.0 |
| DYToLL_M-50_14TeV_pythia8_cff |  |
| DisplacedSUSY_stopToBottom_M_300_1000mm_TuneCUETP8M1_13TeV_pythia8_cff |  |
| EtaBToJpsiJpsi_forSTEAM_TuneCUEP8M1_13TeV_cfi |  |
| GGToHtautau_13TeV_pythia8_cff |  |
| GGToHtautau_13TeV_pythia8_taupinu_cff |  |
| GGToHtautau_13TeV_pythia8_taurhonu_cff |  |
| H125GGgluonfusion_13TeV_TuneCUETP8M1_cfi | 135.7,2017.7,2018.7,25403.0,25403.17,25403.18,250403.0,250403.17,250403.18 |
| H125GGgluonfusion_14TeV_TuneCUETP8M1_cfi |  |
| H130GGgluonfusion_14TeV_TuneCUETP8M1_cfi |  |
| H130GGgluonfusion_8TeV_TuneCUETP8M1_cfi | 5.7 |
| H165WW2L_8TeV_TuneCUETP8M1_cfi | 108.0 |
| H200ZZ4L_13TeV_TuneCUETP8M1_cfi |  |
| H200ZZ4L_8TeV_TuneCUETP8M1_cfi | 109.0 |
| HSCPstop_M_200_TuneCUETP8M1_13TeV_pythia8_cff |  |
| InclusiveppMuX_8TeV_TuneCUETP8M1_cfi | 103.0 |
| JpsiMM_13TeV_TuneCUETP8M1_cfi |  |
| JpsiMM_14TeV_TuneCUETP8M1_cfi |  |
| JpsiMM_8TeV_TuneCUETP8M1_cfi |  |
| JpsiMM_Pt_20_inf_8TeV_TuneCUETP8M1_cfi | 111.0 |
| JpsiMuMu_Pt-8_forSTEAM_13TeV_TuneCUETP8M1_cfi |  |
| MinBias_13TeV_pythia8_TuneCUETP8M1_cfi | 135.8,1301.0,1301.17,2017.8,2018.8 |
| MinBias_13TeV_pythia8_cff | 503.0 |
| MinBias_14TeV_pythia8_TuneCUETP8M1_cfi |  |
| MinBias_8TeV_pythia8_TuneCUETP8M1_cff | 1.0,105.0 |
| MinBias_TuneCUETP8M1_13TeV_pythia8_cff |  |
| PYTHIA8_PhiToMuMu_TuneCUETP8M1_13TeV_cff |  |
| PhotonJet_Pt_10_13TeV_TuneCUETP8M1_cfi |  |
| PhotonJet_Pt_10_14TeV_TuneCUETP8M1_cfi |  |
| PhotonJet_Pt_10_8TeV_TuneCUETP8M1_cfi |  |
| Pythia8_PhotonJetpt15_20_10TeV_cff |  |
| QCDForPF_13TeV_TuneCUETP8M1_cfi | 135.6,2017.6,2018.6,25409.0,25409.17,25409.18,250408.0,250408.17,250408.18 |
| QCDForPF_14TeV_TuneCUETP8M1_cfi |  |
| QCDForPF_8TeV_TuneCUETP8M1_cfi | 5.6 |
| QCD_Pt-15To7000_TuneCUETP8M1_Flat_14TeV-pythia8_cff |  |
| QCD_Pt-20toInf_MuEnrichedPt15_TuneCUETP8M1_13TeV_pythia8_cff |  |
| QCD_Pt-20toInf_MuEnrichedPt15_TuneCUETP8M1_14TeV_pythia8_cff |  |
| QCD_Pt_120_170_13TeV_TuneCUETP8M1_cfi |  |
| QCD_Pt_120_170_8TeV_TuneCUETP8M1_cfi | 117.0 |
| QCD_Pt_15_20_8TeV_TuneCUETP8M1_cfi | 112.0 |
| QCD_Pt_170_230_8TeV_TuneCUETP8M1_cfi | 118.0 |
| QCD_Pt_20_30_8TeV_TuneCUETP8M1_cfi | 113.0 |
| QCD_Pt_3000_3500_13TeV_TuneCUETP8M1_cfi | 1303.0,1303.17 |
| QCD_Pt_3000_3500_14TeV_TuneCUETP8M1_cfi |  |
| QCD_Pt_3000_3500_8TeV_TuneCUETP8M1_cfi | 3.0,123.2 |
| QCD_Pt_30_13TeV_pythia8_cff | 504.0 |
| QCD_Pt_30_50_13TeV_TuneCUETP8M1_cfi |  |
| QCD_Pt_30_50_8TeV_TuneCUETP8M1_cfi | 114.0 |
| QCD_Pt_30_80_BCtoE_8TeV_TuneCUETP8M1_cfi | 4007.0 |
| QCD_Pt_50_80_8TeV_TuneCUETP8M1_cfi | 115.0 |
| QCD_Pt_600_800_13TeV_TuneCUETP8M1_cfi |  |
| QCD_Pt_600_800_14TeV_TuneCUETP8M1_cfi |  |
| QCD_Pt_600_800_8TeV_TuneCUETP8M1_cfi |  |
| QCD_Pt_80_120_13TeV_TuneCUETP8M1_cfi | 135.12,149.0,2017.12,2018.12 |
| QCD_Pt_80_120_14TeV_TuneCUETP8M1_cfi |  |
| QCD_Pt_80_120_8TeV_TuneCUETP8M1_cfi | 116.0,123.1 |
| QCD_Pt_80_170_BCtoE_8TeV_TuneCUETP8M1_cfi | 4008.0 |
| QQH120Inv_8TeV_TuneCUETP8M1_cfi | 107.0 |
| QQH1352T_13TeV_TuneCUETP8M1_cfi |  |
| QQH1352T_8TeV_TuneCUETP8M1_cfi | 34.0,204.0 |
| RS750_quarks_and_leptons_8TeV_TuneCUETP8M1_cff | 133.0 |
| RSGravitonToGammaGamma_kMpl01_M_3000_TuneCUETP8M1_13TeV_pythia8_cfi |  |
| RSKKGluon_m3000GeV_13TeV_TuneCUETP8M1_cff |  |
| RSKKGluon_m3000GeV_13TeV_cff |  |
| SMS-T1tttt_mGl-1500_mLSP-100_13TeV-pythia8_cfi | 135.11,2017.11,2018.11,25408.0,25408.17,25408.18,250409.0,250409.17,250409.18 |
| SoftQCDDiffractive_13TeV_pythia8_cff | 507.0 |
| SoftQCDelastic_13TeV_pythia8_cff | 509.0 |
| SoftQCDinelastic_13TeV_pythia8_cff | 510.0 |
| SoftQCDnonDiffractive_13TeV_pythia8_cff | 508.0 |
| TTbarLepton_13TeV_TuneCUETP8M1_cfi |  |
| TTbarLepton_8TeV_TuneCUETP8M1_cfi |  |
| TTbar_13TeV_TuneCUETP8M1_cfi | 135.1,135.13,1302.0,1302.181,1302.17,1302.18,2017.1,2017.13,2018.1,2018.13,25402.17,25402.0,25402.18,250202.118,250202.183,250202.182,250402.0,250402.17,250402.1,250402.171,250402.18,250402.181 |
| TTbar_14TeV_TuneCUETP8M1_cfi |  |
| TTbar_8TeV_TuneCUETP8M1_cfi | 2.0,5.1,121.0 |
| UpsMM_8TeV_TuneCUETP8M1_cfi | 122.0 |
| Upsilon1SToMuMu_forSTEAM_13TeV_TuneCUETP8M1_cfi |  |
| WE_13TeV_TuneCUETP8M1_cfi |  |
| WE_14TeV_TuneCUETP8M1_cfi |  |
| WE_8TeV_TuneCUETP8M1_cfi |  |
| WM_13TeV_TuneCUETP8M1_cfi |  |
| WM_14TeV_TuneCUETP8M1_cfi |  |
| WM_8TeV_TuneCUETP8M1_cfi |  |
| WToLNu_13TeV_pythia8_cff | 506.0 |
| Wjet_Pt_3000_3500_13TeV_TuneCUETP8M1_cfi |  |
| Wjet_Pt_3000_3500_14TeV_TuneCUETP8M1_cfi |  |
| Wjet_Pt_3000_3500_8TeV_TuneCUETP8M1_cfi |  |
| Wjet_Pt_80_120_13TeV_TuneCUETP8M1_cfi |  |
| Wjet_Pt_80_120_14TeV_TuneCUETP8M1_cfi |  |
| Wjet_Pt_80_120_8TeV_TuneCUETP8M1_cfi |  |
| WpM_13TeV_TuneCUETP8M1_cfi |  |
| WpM_8TeV_TuneCUETP8M1_cfi |  |
| WprimeToENu_M-2000_TuneCUETP8M1_13TeV-pythia8_cff |  |
| WprimeTolNu_M3000_13TeV_pythia8_cff | 561.0 |
| ZEEMM_13TeV_TuneCUETP8M1_cfi |  |
| ZEE_13TeV_TuneCUETP8M1_cfi | 135.4,1304.18,1304.181,1304.182,1356.0,2017.4,2018.4,25400.0,25400.17,25400.18,250200.1,250200.117,250200.118,250200.182,250400.0,250400.17,250400.18,500200.1 |
| ZEE_14TeV_TuneCUETP8M1_cfi |  |
| ZEE_8TeV_TuneCUETP8M1_cfi | 5.4 |
| ZMM_13TeV_TuneCUETP8M1_cfi | 135.9,2017.9,2018.9,25406.0,25406.17,25406.18,250206.118,250206.182,250406.0,250406.17,250406.18 |
| ZMM_14TeV_TuneCUETP8M1_cfi |  |
| ZMM_8TeV_TuneCUETP8M1_cfi |  |
| ZPrime5000JJ_8TeV_TuneCUETP8M1_cfi | 125.0 |
| ZTT_All_hadronic_13TeV_TuneCUETP8M1_cfi | 135.5,2017.5,2018.5,25405.0,25405.18,25405.17,250405.17,250405.0,250405.18 |
| ZTT_All_hadronic_8TeV_TuneCUETP8M1_cfi | 119.0 |
| ZmumuJets_Pt_20_300_GEN_13TeV_TuneCUETP8M1_cfg |  |
| ZpEE_2250_13TeV_Tauola_TuneCUETP8M1_cfi |  |
| ZpEE_2250_13TeV_TuneCUETP8M1_cfi |  |
| ZpEE_2250_8TeV_TuneCUETP8M1_cfi |  |
| ZpMM_13TeV_TuneCUETP8M1_cfi |  |
| ZpMM_2250_13TeV_Tauola_TuneCUETP8M1_cfi |  |
| ZpMM_2250_13TeV_TuneCUETP8M1_cfi |  |
| ZpMM_2250_8TeV_TuneCUETP8M1_cfi |  |
| ZpMM_8TeV_TuneCUETP8M1_cfi |  |
| ZpTT_1500_13TeV_TuneCUETP8M1_cfi |  |
| ZpTT_1500_8TeV_TuneCUETP8M1_cfi |  |
| ZprimeToll_M3000_13TeV_pythia8_cff | 560.0 |
| bJpsiX_8TeV_TuneCUETP8M1_cfi | 106.0 |

 4. *added* cfi/cff files for ExternalGeneratorFilter

| Config filename | Matched RelVal ID(s) |
| :--- | :--- |
| BuToKstarJPsiToMuMu_forSTEAM_13TeV_ExtGen_cfi | 570 |
| BsToMuMu_forSTEAM_13TeV_ExtGen_cfi | 571 |
| ZTT_Tauola_OneLepton_OtherHadrons_8TeV_TuneCUETP8M1_ExtGen_cfi | 572 |
| sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen_cff | 573 |
| Hydjet_Quenched_B12_5020GeV_ExtGen_cfi | 574 |
| AMPT_PPb_5020GeV_MinimumBias_ExtGen_cfi | 575 |
| ReggeGribovPartonMC_EposLHC_4080_4080GeV_pPb_ExtGen_cfi | 576 |
| Pyquen_ZeemumuJets_pt10_2760GeV_ExtGen_cfi | 577 |

</details>

tvami commented:
test parameters:
 - enable_test = threading
 - workflow_threading = 570,571,572,573,574,575,576,577,562.0,537.0,1361.17,1361.0,1310.0,560.0
 - workflows = 570,571,572,573,574,575,576,577,562.0,537.0,1361.17,1361.0,1310.0,560.0

tvami commented:
@cmsbuild , please test

colizz committed:
Fix the regex to match the LHE tags in mergeLHE.py

cmsbuild commented:
Pull request #34527 was updated. @SiewYan, @mkirsano, @jordan-martins, @chayanit, @bbilin, @wajidalikhan, @Saptaparna, @cmsbuild, @adrianodee, @srimanob, @agrohsje, @kpedro88, @kskovpen, @alberto-sanchez, @santocch, @GurpreetSinghChahal can you please check and sign again.

colizz commented:
sorry, I forgot to push another backport from #34804. Please help to test again.

(These are all the backports for this thread. PR description will provide a summary)

tvami commented:
@cmsbuild , please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-850492/17588/summary.html
**COMMIT**: b4af9f8dfa0603917a86d27d22302d952bd3b1d4
**CMSSW**: CMSSW_10_6_X_2021-08-01-0000/slc7_amd64_gcc700
**Additional Tests**: THREADING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/34527/17588/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



The workflows 140.53 have different files in step1_dasquery.log than the ones found in the baseline. You may want to check and retrigger the tests if necessary. You can check it in the "files" directory in the results of the comparisons

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/1310.0_ADDMonoJet_d3MD3_13+ADDMonoJet_d3MD3_13+DIGIUP15+RECOUP15+HARVESTUP15
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/1361.0_GluGluHToZZTo4L_M125_Pow_py8_Evt_13+GluGluHToZZTo4L_M125_Pow_py8_Evt_13+DIGIUP15+RECOUP15+HARVESTUP15
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/1361.17_GluGluHToZZTo4L_M125_Pow_py8_Evt_13UP17+GluGluHToZZTo4L_M125_Pow_py8_Evt_13UP17+DIGIUP17+RECOUP17+HARVESTUP17
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/537.0_DYToLL012Jets_5FS_TuneCH3_13TeV_amcatnloFxFx_herwig7+DYToLL012Jets_5FS_TuneCH3_13TeV_amcatnloFxFx_herwig7+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/560.0_ZprimeToll_M3000_13TeV_pythia8+ZprimeToll_M3000_13TeV_pythia8+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/562.0_BulkG_ZZ_2L2Q_M1200_narrow_13TeV_pythia8+BulkG_M1200_narrow_2L2Q_LHE_13TeV+Hadronizer_TuneCUETP8M1_Mad_pythia8+HARVESTGEN2
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/570.0_BuToKstarJPsiToMuMu_forSTEAM_13TeV_ExtGen+BuToKstarJPsiToMuMu_forSTEAM_13TeV_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/571.0_BsToMuMu_forSTEAM_13TeV_ExtGen+BsToMuMu_forSTEAM_13TeV_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/572.0_ZTTFS_ExtGen+ZTTFS_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/573.0_sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen+sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/574.0_HydjetQ_B12_5020GeV_2018_ExtGen+HydjetQ_B12_5020GeV_2018_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/575.0_AMPT_PPb_5020GeV_MinimumBias_ExtGen+AMPT_PPb_5020GeV_MinimumBias_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/576.0_EPOS_PPb_8160GeV_MinimumBias_ExtGen+EPOS_PPb_8160GeV_MinimumBias_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/577.0_Pyquen_ZeemumuJets_pt10_2760GeV_ExtGen+Pyquen_ZeemumuJets_pt10_2760GeV_ExtGen+HARVESTGEN

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 1240 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 3215686
 - DQMHistoTests: Total failures: 2040
 - DQMHistoTests: Total nulls: 22
 - DQMHistoTests: Total successes: 3213290
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -56.705 KiB( 34 files compared)
 - DQMHistoSizes: changed ( 140.53 ): -44.531 KiB      Hcal/DigiRunHarvesting
 - DQMHistoSizes: changed ( 140.53 ): -10.938 KiB      Info/EventInfo
 - DQMHistoSizes: changed ( 140.53 ): -1.172 KiB       RPC/DCSInfo
 - DQMHistoSizes: changed ( 140.53 ): -0.064 KiB       SiStrip/MechanicalView
 - Checked 143 log files, 29 edm output root files, 35 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_6_X_2021-08-01-0000+850492/44635/triggerResults): no differences found

agrohsje commented:
+generators

jordan-martins commented:
+1

srimanob commented:
+Upgrade

agrohsje commented:
@qliphy @colizz :
as there will be no pdd this week (most likely) congqiao will present in next weeks ppd meeting.

jordan-martins commented:
Hi @colizz and @qliphy , do we know when we do expect this PR to be merged?
Thanks!

qliphy commented:
@jordan-martins  The PR (https://github.com/cms-sw/cmssw/pull/34710) in the master will be merged soon this week.  After checking in master IB,  @colizz will report at PPD on Aug 26th, and then we will merge this one (and start building a new 10_6_X release) if everything looks ok.

perrotta commented:
backport of #34710 
(as well as of several other older PRs, as listed in the PR desctiption)
This is to make explicitly visible the dependency on the still to be merged in master PR

perrotta commented:
@colizz please list in the PR description also the dependency from #34710, so that it will remain there for reference

colizz commented:
> @colizz please list in the PR description also the dependency from #34710, so that it will remain there for reference

Thanks @perrotta , these are mentioned under bullet-7 in the PR description - I expend the item for clarity.

qliphy commented:
please abort
(to try again with a new IB including a fix with https://github.com/cms-sw/cmssw/pull/35047)

qliphy commented:
test parameters:

qliphy commented:
test parameters:
- enable_test = threading
- workflow_threading = 570,571,572,573,574,575,576,577,562.0,537.0,1361.17,1361.0,1310.0,560.0
- workflows = 570,571,572,573,574,575,576,577,562.0,537.0,1361.17,1361.0,1310.0,560.0

qliphy commented:
please test

qliphy commented:
@colizz  A 2nd run shows there is still an issue with workflow 573:
https://cmssdt.cern.ch/jenkins/job/ib-run-pr-relvals/9601/console
19:18:46 573.0_sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen+sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen+HARVESTGEN Step0-FAILED Step1-NOTRUN  - time date Sat Aug 28 12:55:06 2021-date Sat Aug 28 12:34:10 2021; exit: 16640 0

Should the following line be adjusted?  
https://github.com/cms-sw/cmssw/blob/b4af9f8dfa0603917a86d27d22302d952bd3b1d4/Configuration/Generator/python/sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen_cff.py#L66
Or we leave the change on the file from this PR?

colizz commented:
> @colizz A 2nd run shows there is still an issue with workflow 573:
> https://cmssdt.cern.ch/jenkins/job/ib-run-pr-relvals/9601/console
> 19:18:46 573.0_sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen+sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen+HARVESTGEN Step0-FAILED Step1-NOTRUN - time date Sat Aug 28 12:55:06 2021-date Sat Aug 28 12:34:10 2021; exit: 16640 0
> 
> Should the following line be adjusted?
> https://github.com/cms-sw/cmssw/blob/b4af9f8dfa0603917a86d27d22302d952bd3b1d4/Configuration/Generator/python/sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen_cff.py#L66
> 
> 
> Or we leave the change on the file from this PR?

Thanks @qliphy , indeed this line need to be removed (i.e. we don't need to extend the waitTime from 300s to 1200s), because my previous interpretation of this error is wrong. 
My current investigation is that this is still a physics issue caused by generating 0 events. More study is underway.

colizz commented:
Hello,

For the error of workflow 573, my conclusion is that extending the event number from 10 to 50 may solve the issue. Here is the story:

The error (@qliphy pointed out here https://github.com/cms-sw/cmssw/pull/34527#issuecomment-906996422) happens in the endLumi phase in one working thread. It is only randomly triggered when the thread is unlucky that it receives zero events in the event loop.

The reason that we encounter some threads handles zero events is because (1) event processing in the threading framework follows the first-come-fisrt-serve principle, and (2) the threads (let's say 4 threads in total) are unsynchronous in accepting the first event, because they finish the beginLumi phase not at the same time - which is mainly brought by the mechanism of the ExtenralGeneratorFilter that stream0 must run beginLumi first, then followed by stream1/2/3 (thus stream0 will start accepting events earlier than stream1/2/3). 

In this case, when requesting 10 events in this workflow, the slowest thread will probabilistically miss all events (they are all assigned to other threads), and will directly call the "event summary" module in sherpa just after its initialization, causing this crash. However, not all threads handling zero events will crash. The true mechanism is yet to be discovered... (probably need to consult Sherpa expert)

I think we may extend 10 events to 50 temporarily. My existing test shows
1. Launching this workflow with 50 events on a busy lxplus machine (busy machine means slower beginLumi processing, hence more unsynchronization happened among the 4 threads to start accepting their first events, resulting larger probability to reproduce the issue). I launch 100 times and see no such error occurring. As a comparison, launching 10 events in the same environment has 1/4 falling probability.
2. In the 10_6_X UL sample validation, we already examine the sherpa+ExternalGeneratorFilter configurations in the following sherpa samples
```
SMP-RunIISummer20UL16GEN-00007
SMP-RunIISummer20UL16GENAPV-00005
SMP-RunIISummer20UL17GEN-00011
SMP-RunIISummer20UL18GEN-00006
HIG-RunIISummer19UL17GEN-00004
HIG-RunIISummer19UL17GEN-00006
SMP-RunIISummer19UL17GEN-00005
SMP-RunIISummer19UL17GEN-00006
```

The physics results agree well and we do not see such a bug. The reason can easily be that we are launching many events (~1000 events) on 4 threads. The number is so large that it can be guaranteed no thread will be unlucky to handle zero events.

My plan is to simply change the event number to 50 in a master PR on Monday, if no objection. 
Please let me know if you have suggestions. Many thanks！

(FYI: @agrohsje @GurpreetSinghChahal @Saptaparna @qliphy )

jordan-martins commented:
Hi @colizz , your assessment seems reasonable indeed. If no constraint from the 50 number from @qliphy and @perrotta, then lets move it.
Thanks a lot!

qliphy commented:
> Hi @colizz , your assessment seems reasonable indeed. If no constraint from the 50 number from @qliphy and @perrotta, then lets move it.
 
I am fine with this.

colizz committed:
Fix the random segfault of RelVal workflow 573 (10_6_X)

Backport of #35072

colizz commented:
The change is done (PR of #35072), please help to test again, thanks!

qliphy commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-850492/18140/summary.html
**COMMIT**: 0cf3e81834b8aca603a386ddcbe7db72e4e73f86
**CMSSW**: CMSSW_10_6_X_2021-08-30-1100/slc7_amd64_gcc700
**Additional Tests**: THREADING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/34527/18140/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/1310.0_ADDMonoJet_d3MD3_13+ADDMonoJet_d3MD3_13+DIGIUP15+RECOUP15+HARVESTUP15
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/1361.0_GluGluHToZZTo4L_M125_Pow_py8_Evt_13+GluGluHToZZTo4L_M125_Pow_py8_Evt_13+DIGIUP15+RECOUP15+HARVESTUP15
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/1361.17_GluGluHToZZTo4L_M125_Pow_py8_Evt_13UP17+GluGluHToZZTo4L_M125_Pow_py8_Evt_13UP17+DIGIUP17+RECOUP17+HARVESTUP17
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/537.0_DYToLL012Jets_5FS_TuneCH3_13TeV_amcatnloFxFx_herwig7+DYToLL012Jets_5FS_TuneCH3_13TeV_amcatnloFxFx_herwig7+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/560.0_ZprimeToll_M3000_13TeV_pythia8+ZprimeToll_M3000_13TeV_pythia8+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/562.0_BulkG_ZZ_2L2Q_M1200_narrow_13TeV_pythia8+BulkG_M1200_narrow_2L2Q_LHE_13TeV+Hadronizer_TuneCUETP8M1_Mad_pythia8+HARVESTGEN2
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/570.0_BuToKstarJPsiToMuMu_forSTEAM_13TeV_ExtGen+BuToKstarJPsiToMuMu_forSTEAM_13TeV_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/571.0_BsToMuMu_forSTEAM_13TeV_ExtGen+BsToMuMu_forSTEAM_13TeV_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/572.0_ZTTFS_ExtGen+ZTTFS_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/573.0_sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen+sherpa_ttbar_2j_MENLOPS_13TeV_MASTER_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/574.0_HydjetQ_B12_5020GeV_2018_ExtGen+HydjetQ_B12_5020GeV_2018_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/575.0_AMPT_PPb_5020GeV_MinimumBias_ExtGen+AMPT_PPb_5020GeV_MinimumBias_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/576.0_EPOS_PPb_8160GeV_MinimumBias_ExtGen+EPOS_PPb_8160GeV_MinimumBias_ExtGen+HARVESTGEN
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-850492/577.0_Pyquen_ZeemumuJets_pt10_2760GeV_ExtGen+Pyquen_ZeemumuJets_pt10_2760GeV_ExtGen+HARVESTGEN

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 3215686
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3215351
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 143 log files, 29 edm output root files, 35 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_6_X_2021-08-30-1100+850492/45077/triggerResults): no differences found

qliphy commented:
Looks good now.  thanks @colizz 

@cms-sw/pdmv-l2 @cms-sw/generators-l2 @cms-sw/upgrade-l2  Would you please check and sign again?

agrohsje commented:
+generators
Should find a better solution for the first come first serve problem but I am happy with the current fix. Thanks @colizz .

AdrianoDee commented:
+upgrade
Fix from @colizz working smoothly for the threading issue. Thanks.

jordan-martins commented:
+1

qliphy commented:
+1

qliphy commented:
merge

santocch commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_10_6_X IBs (tests are also fine) and once validation in the development release cycle CMSSW_12_1_X is complete. This pull request will be automatically merged.
